src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -121,32 +134,156 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
               '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[4])].trim(),
         })));
 
-    this.studentService.enrollStudents(
-        this.courseId, studentsEnrollRequest,
-    ).pipe(finalize(() => this.isEnrolling = false)).subscribe((resp: Students) => {
-      const enrolledStudents: Student[] = resp.students;
-      this.showEnrollResults = true;
-      this.statusMessage.pop(); // removes any existing error status message
-      this.statusMessageService.showSuccessToast('Enrollment successful. Summary given below.');
-      this.enrollResultPanelList =
-          this.populateEnrollResultPanelList(this.existingStudents, enrolledStudents,
-              studentsEnrollRequest.studentEnrollRequests);
-    }, (resp: ErrorMessageOutput) => {
-      this.enrollErrorMessage = resp.error.message;
-    }, () => {
-      this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
-        this.existingStudents = resp.students;
-        if (!this.isExistingStudentsPanelCollapsed) {
-          const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
-          this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
+    this.highLightSectionsNull(studentsEnrollRequest);
```

Comment:
no need for so many functions - just have three (use the function naming below):
- `checkEmailNotRepeated`: checks emails are not repeated across table
- `checkTeamsValid`: checks teams are not repeated across section
- `checkCompulsoryField`: checks all compulsory fields 

Update the error messages accordingly as well - let's try to be as clear as possible to the end users.  We probably should add a new message for repeated emails e.g. "Duplicate email addresses detected."